src/EnergyPlus/Vectorize.hh

```diff
@@ -0,0 +1,34 @@
+#ifndef EnergyPlus_vectorize_hh_INCLUDED
+#define EnergyPlus_vectorize_hh_INCLUDED
+
+// EnergyPlus Vectorization Support
+
+// ObjexxFCL Headers
+#include <ObjexxFCL/vectorize.hh> // ASSUME_ALIGNED and RESTRICT macros
+
+// C++ Headers
+#include <cassert>
+#include <cstdint>
+
+// EnergyPlus headers
+#include "EnergyPlus.hh"
+
+namespace EnergyPlus {
+
+	int const VEC_LENGTH = VEC_ALIGN / sizeof(Real64);
+
+#define ROUND_TO_VEC_LENGTH(I) (((I) + ((VEC_LENGTH) - 1)) & ~((VEC_LENGTH) - 1))
+
+
+#define EXPLICIT_VECTORIZATION
+
+	// EXPLICIT_VECTORIZATION assumes that vectors are aligned,
+	// which is not a valid assumption on 32-bit builds
+#if defined(_WIN32) && !defined(_WIN64)
```

Comment:
@myoldmopar, please hold off on PR'ing this. There are still a small number
of "big diffs" that need to be investigated.

On Mon, Dec 14, 2015 at 8:20 AM, Edwin Lee <notifications@github.com> wrote:

> In src/EnergyPlus/Vectorize.hh
> <https://github.com/NREL/EnergyPlus/pull/5278#discussion_r47495932>:
>
> > +
> > +// EnergyPlus headers
> > +#include "EnergyPlus.hh"
> > +
> > +namespace EnergyPlus {
> > +
> > +	int const VEC_LENGTH = VEC_ALIGN / sizeof(Real64);
> > +
> > +#define ROUND_TO_VEC_LENGTH(I) (((I) + ((VEC_LENGTH) - 1)) & ~((VEC_LENGTH) - 1))
> > +
> > +
> > +#define EXPLICIT_VECTORIZATION
> > +
> > +	// EXPLICIT_VECTORIZATION assumes that vectors are aligned,
> > +	// which is not a valid assumption on 32-bit builds
> > +#if defined(_WIN32) && !defined(_WIN64)
>
> EnergyPlus is running slow on my 386 help!
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/NREL/EnergyPlus/pull/5278/files#r47495932>.
>